Bottom: b3b7b86dfc08a36e635244eb82af20102e9fbdff
Top:    0ed193b0973e89b5780d033a25f8773fd28fa822
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-25 15:47:51 -0500

Refresh of tmp-mantis-database

---
diff --git a/roles/docker-mantisbt/defaults/main.yml b/roles/docker-mantisbt/defaults/main.yml
index 7e8179c4..61e5ff0c 100644
--- a/roles/docker-mantisbt/defaults/main.yml
+++ b/roles/docker-mantisbt/defaults/main.yml
@@ -24,8 +24,8 @@ mantisbt_db_user: novatech
 mantisbt_image_repo: https://github.com/novatechweb/docker-mantisbt.git
 mantisbt_image_name: '{{ docker_registry_username }}/mantisbt'
 
-mantisbt_db_image_repo: https://github.com/novatechweb/docker-mysql-data.git
-mantisbt_db_image_name: '{{ docker_registry_username }}/mysql.dv'
+mantisbt_db_image_name: 'mysql'
+mantisbt_db_image_tag: 5
 
 # the name of the container being started
 mantisbt_container_name: '{{ docker_name_prefix }}mantisbt'
@@ -36,6 +36,9 @@ mantisbt_db_container_name: '{{ docker_name_prefix }}mantisbt_db'
 # the name of the data-volume used by the container
 mantisbt_dv_name: '{{ docker_name_prefix }}mantisbt_DV'
 
+# the name of the mysql config volume used by the mantisbt container
+mantisbt_db_cv_name: '{{ docker_name_prefix }}mantisbt_db_CV'
+
 # the name of the mysql data-volume used by the mantisbt container
 mantisbt_db_dv_name: '{{ docker_name_prefix }}mantisbt_db_DV'
 
@@ -47,3 +50,9 @@ mantisbt_docker_restore_dir: '{{ bacula_dest }}{{ mantisbt_docker_backup_dir }}'
 mantisbt_static_backup_file: '/mantisbt.tar'
 mantisbt_database_backup_file: '/mantisbtdb.sql'
 mantisbt_database_files_table_backup_file: '/mantisbtdb-files_table.sql'
+
+# Network names
+docker_network_frontend: 'frontend'
+
+# the name of the sql database used by mantisbt
+mantisbt_database_name: 'bugtracker'
diff --git a/roles/docker-mantisbt/tasks/main.yml b/roles/docker-mantisbt/tasks/main.yml
index 45c1a895..478f6a6d 100644
--- a/roles/docker-mantisbt/tasks/main.yml
+++ b/roles/docker-mantisbt/tasks/main.yml
@@ -85,30 +85,6 @@
     image_tag: '{{ docker_image_tag }}'
     dockerfile_dir: '{{ docker_projects_dir }}/docker-mantisbt'
 
-- name: Checkout db repo
-  git:
-    repo: '{{ mantisbt_db_image_repo }}'
-    version: master
-    dest: '{{ docker_projects_dir }}/docker-mysql-data'
-
-- name: build data-container image
-  docker_build:
-    image_name: '{{ mantisbt_db_image_name }}'
-    image_tag: '{{ docker_image_tag }}'
-    dockerfile_dir: '{{ docker_projects_dir }}/docker-mysql-data'
-
-# *****************************************************************************
-# copy the MySQL config to the restore directory
-
-- name: copy mysql config
-  copy:
-    src: '{{ item }}'
-    dest: '{{ host_mantisbt_docker_restore_dir }}/{{ item }}'
-    setype: svirt_sandbox_file_t
-  with_items:
-  - largefile.cnf
-  - character-set.cnf
-
 # *****************************************************************************
 # Create the data volumes
 
@@ -118,63 +94,86 @@
     image_tag: '{{ docker_image_tag }}'
     data_volume_container_name: '{{ mantisbt_dv_name }}'
 
+- name: config-volume container (mysql)
+  docker_volume:
+    name: '{{ mantisbt_db_cv_name }}'
+
 - name: data-volume container (mysql)
   docker_datavolume:
     image_name: '{{ mantisbt_db_image_name }}'
     image_tag: '{{ docker_image_tag }}'
     data_volume_container_name: '{{ mantisbt_db_dv_name }}'
 
-- name: stop prev container (mysql)
-  docker_container:
-    image: '{{ mantisbt_db_image_name }}:{{ docker_image_tag }}'
-    name: '{{ mantisbt_db_container_name }}'
-    state: stopped
-
-- name: initial populate (mysql - import config)
-  command: 'docker run --rm --volumes-from {{ mantisbt_db_dv_name }} -v {{ host_mantisbt_docker_restore_dir }}:/tmp/conf.d {{ mantisbt_db_image_name }}:{{ docker_image_tag }}'
+- name: make temporary directory for config
+  tempfile:
+    state: directory
+  register: config_tempdir
 
-- name: initial populate (mysql - database)
-  docker_db_init:
-    image_name: '{{ mantisbt_db_image_name }}'
-    image_tag: '{{ docker_image_tag }}'
-    container_name: '{{ mantisbt_db_dv_name }}_init_db'
-    data_volume_container_name: '{{ mantisbt_db_dv_name }}'
-    database_name: bugtracker
-    database_user: '{{ mantisbt_db_user }}'
-    database_password: '{{ mantisbt_db_password | quote }}'
-    database_root_password: '{{ mantisbt_db_root_password | quote }}'
+- name: upload mysql config
+  copy:
+    src: '{{ item }}'
+    dest: '{{ config_tempdir.path }}/{{ item }}'
+  with_items:
+  - largefile.cnf
+  - character-set.cnf
 
-# *****************************************************************************
-# Start the data container running
+- name: copy mysql config to volume
+  command: >
+    docker run --rm 
+    -v '{{ config_tempdir.path }}:/from'
+    -v '{{ mantisbt_db_cv_name }}:/to'
+    alpine /bin/ash -c 'cp -v /from/* /to'
 
 - name: start container (mysql)
   docker_container:
-    name: '{{ mantisbt_db_container_name }}'
+    name: '{{ mantisbt_db_container_name}}'
     detach: true
     restart_policy: '{{ docker_restart_policy }}'
-    volumes_from:
-      - '{{ mantisbt_db_dv_name }}'
-    image: '{{ mantisbt_db_image_name }}:{{ docker_image_tag }}'
+    image: '{{ mantisbt_db_image_name }}'
+    volumes:
+      - '{{ mantisbt_db_cv_name }}:/etc/mysql/conf.d:z'
+      - '{{ mantisbt_db_dv_name }}:/var/lib/mysql:z'
+    env:
+      MYSQL_DATABASE: '{{ mantisbt_database_name }}'
+      MYSQL_ROOT_PASSWORD: '{{ mantisbt_db_root_password }}'
+      MYSQL_USER: '{{ mantisbt_db_user }}'
+      MYSQL_PASSWORD: '{{ mantisbt_db_password }}'
+    networks:
+      - name: '{{ docker_network_frontend }}'
+
+- name: wait for initialization to complete
+  shell: >
+      printf 'SHOW GLOBAL STATUS\n' | docker exec -i 
+      '{{ mantisbt_db_container_name}}'
+      mysql --host=localhost 
+      --user={{ mantisbt_db_user | quote }}
+      --password={{ mantisbt_db_password | quote }} 
+      '{{ mantisbt_database_name }}'
+  register: db_status
+  retries: 20
+  delay: 3
+  until: "db_status.rc == 0"
+
+# *****************************************************************************
+# Start the data container running
 
 - name: start container (mantisbt)
   docker_container:
     name: '{{ mantisbt_container_name }}'
     detach: true
     restart_policy: '{{ docker_restart_policy }}'
-    hostname: '{{ container_addr_map.mantisbt.hostname }}'
-    ports: '{{ container_port_map.mantisbt.port_args }}'
-    link: 
-      -'{{ openldap_container_name }}:mantisbt_ldap'
-      -'{{ mantisbt_db_container_name }}:mantisbt_db'
-      -'{{ exim4_container_name }}:{{ container_addr_map.exim4.hostname }}'
-    volumes_from: 
-      - '{{ openssl_dv_name }}'
-      - '{{ mantisbt_dv_name }}'
+    hostname: '{{ mantisbt_hostname }}'
+    ports: '{{ mantisbt_port_args }}'
+    networks:
+      - name: '{{ docker_network_frontend }}'
+    volumes: 
+      - '{{ openssl_dv_name }}:/etc/ssl/private:z'
+      - '{{ mantisbt_dv_name }}:/var/www/html:z'
     env: 
-      - MANTISBT_HOSTNAME={{ container_addr_map.mantisbt.hostname }}
-      - MANTISBT_MAIL_USER=mantis
-      - MANTISBT_MAIL_PASSWORD={{ mantisbt_email_password | quote }}
-      - MANTISBT_LDAP_PASSWORD={{ ldap_proxyagent_password | quote }}
+      MANTISBT_HOSTNAME: '{{ mantisbt_hostname }}'
+      MANTISBT_MAIL_USER: 'mantis'
+      MANTISBT_MAIL_PASSWORD: '{{ mantisbt_email_password | quote }}'
+      MANTISBT_LDAP_PASSWORD: '{{ openldap_proxyagent_password | quote }}'
     image: '{{ mantisbt_image_name }}:{{ docker_image_tag }}'
 
 # *****************************************************************************
diff --git a/roles/docker-mantisbt/templates/config.sh.j2 b/roles/docker-mantisbt/templates/config.sh.j2
index f9538c8a..6f4f7087 100644
--- a/roles/docker-mantisbt/templates/config.sh.j2
+++ b/roles/docker-mantisbt/templates/config.sh.j2
@@ -11,5 +11,5 @@
   STATIC_BACKUP_FILE='{{ mantisbt_static_backup_file }}'
   DATABASE_BACKUP_FILE='{{ mantisbt_database_backup_file }}'
   DATABASE_FILES_TABLE_BACKUP_FILE='{{ mantisbt_database_files_table_backup_file }}'
-  MANTISBT_DB_DB_NAME='bugtracker'
+  MANTISBT_DB_DB_NAME='{{ mantisbt_database_name }}'
 }
